File: python/algo/202606/algoiv01.08.py
from typing import List


class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        m, n = len(matrix), len(matrix[0])

        rows, cols = set(), set()
        for i, row in enumerate(matrix):
            for j, v in enumerate(row):
                if v == 0:
                    rows.add(i)
                    cols.add(j)
        
        for r in rows:
            for j in range(n):
                matrix[r][j] = 0
        for c in cols:
            for i in range(m):
                matrix[i][c] = 0

File: cplus/leetcode/202606/algoiv01.08.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    void setZeroes(vector<vector<int>>& matrix) {
        int m = matrix.size(), n = matrix[0].size();
        set<int> rows, cols;
        for (int i=0; i<m; i++) for (int j=0; j<n; j++) {
            if (matrix[i][j] == 0) {
                rows.insert(i); cols.insert(j);
            }
        }

        for (auto r: rows) for (int j=0; j<n; j++) matrix[r][j] = 0;
        for (auto c: cols) for (int i=0; i<m; i++) matrix[i][c] = 0;
    }
};